core/src/builtin_widgets/scrollable.rs

```diff
@@ -14,6 +14,35 @@ pub enum Scrollable {
   Both,
 }
 
+pub struct ScrollViewInfo {
+  pub scroll_pos: Point,
+  pub view_size: Size,
+
+  scroll_id: TrackId,
+  wnd: Sc<Window>,
+}
+
+impl ScrollViewInfo {
+  pub fn map_to_scroll_view(&self, p: Point, from: WidgetId) -> Point {
+    let pos = self.wnd.map_to_global(p, from);
+    let offset = self
+      .wnd
+      .map_to_global(Point::zero(), self.scroll_id.get().unwrap());
+    pos - offset.to_vector()
+  }
+}
+
+pub struct ScrollRequest(Box<dyn FnMut(ScrollViewInfo) -> Point>);
```

Comment:
Perhaps we could consider offering a `Scrollable` widget instead of relying on custom events to control the `ScrollableWidget`.